dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3693,25 +3696,38 @@ Sedp::TypeLookupReplyReader::~TypeLookupReplyReader()
 }
 
 DDS::ReturnCode_t
-Sedp::TypeLookupRequestWriter::send_type_lookup_request(XTypes::TypeIdentifierSeq& type_ids,
-                                                        const DCPS::RepoId& reader,
-                                                        DCPS::SequenceNumber& sequence,
-                                                        const DCPS::SequenceNumber& rpc_sequence,
-                                                        const DCPS::RepoId& participant_id)
+Sedp::TypeLookupRequestWriter::send_type_lookup_request(const XTypes::TypeIdentifierSeq& type_ids,
+  const DCPS::RepoId& reader,
+  DCPS::SequenceNumber& sequence,
+  const DCPS::SequenceNumber& rpc_sequence,
+  const DCPS::RepoId& participant_id,
+  const CORBA::ULong tl_kind)
 {
-  XTypes::TypeLookup_Request type_lookup_request;
-
-  type_lookup_request.data.getTypes.type_ids = type_ids;
-  type_lookup_request.data.kind = XTypes::TypeLookup_getTypes_HashId;
+  if (tl_kind != XTypes::TypeLookup_getTypes_HashId &&
+      tl_kind != XTypes::TypeLookup_getDependencies_HashId) {
+    return DDS::RETCODE_BAD_PARAMETER;
+  }
 
+  XTypes::TypeLookup_Request type_lookup_request;
   type_lookup_request.header.request_id.writer_guid = get_repo_id();
-
   type_lookup_request.header.request_id.sequence_number.high = rpc_sequence.getHigh();
   type_lookup_request.header.request_id.sequence_number.low = rpc_sequence.getLow();
 
-  // As per chapter 7.6.3.3.4 of XTypes spec
-  const OPENDDS_STRING instance_name = OPENDDS_STRING("dds.builtin.TOS.") + DCPS::to_string(participant_id);
+  // As per chapter 7.6.3.3.4 of XTypes spec.
+  const OPENDDS_STRING instance_name = OPENDDS_STRING("dds.builtin.TOS.") +
```

Comment:
Is there already code for checking the `instance_name` of an incoming request?

Reply:
No, we haven't done anything with `instance_name` of incoming request. The spec doesn't say how it is handled as far as I know.